manifests/selboolean.pp

```diff
@@ -0,0 +1,20 @@
+# == Define: types::selboolean
+#
+define types::selboolean (
+  $persistent = false,
+  $value      = undef,
+  $provider   = undef,
+) {
+
+  validate_bool($persistent)
+  validate_string($value)
```

Comment:
Thinking it through and looking over the doc, I don't think provider is needed - there is only one valid provider. I'll omit from my next commit.